src/pkg/utils/network.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+func IsUrl(source string) bool {
+	parsedUrl, err := url.Parse(source)
+	return err == nil && parsedUrl.Scheme != "" && parsedUrl.Host != ""
+}
+
+// DoesHostnamesMatch returns a boolean indicating if the hostname of two different URLs are the same.
+func DoesHostnamesMatch(url1 string, url2 string) (bool, error) {
```

Comment:
I know this didn't change but this seems like a weirdly worded function name. Maybe DoHostnamesMatch or AreHostnamesMatch (given it returns a boolean)?